src/test/socket/connect/test_connect.rs

```diff
@@ -318,7 +318,7 @@ fn test_null_addr() -> Result<(), String> {
     let args = ConnectArguments {
         fd: fd,
         addr: None,
-        addr_len: 5,
+        addr_len: std::mem::size_of::<libc::sockaddr_in>() as u32,
     };
```

Comment:
> I think we've agreed before that in such cases where multiple errors could reasonably be returned, particularly when they both indicate programming errors, we wouldn't make an effort to follow which one Linux happens to return.

I do like returning the same error as Linux just because it makes the tests easier to write, but I don't think having a different error should ever block us from merging something.

> I could potentially revert this change and change the syscall handling to ensure EFAULT gets returned in this case. In that case I think this test would be more accurately something like test_null_and_bad_addrlen; maybe it could be split into two tests. I still don't really think it's worth it, though.

Current PR version is good with me!